reactor-netty-core/src/main/java/reactor/netty/transport/ClientTransport.java

```diff
@@ -250,6 +251,45 @@ public T proxy(Consumer<? super ProxyProvider.TypeSpec> proxyOptions) {
 		return dup;
 	}
 
+	/**
+	 * Set up proxy from java system properties.
+	 * Supports http, https, socks4, socks5 proxies.
+	 * List of supported system properties https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+	 *
+	 * If both {@code https.proxyHost} and {@code http.proxyHost} are set
+	 * it chooses {@code https.proxyHost} over {@code http.proxyHost}.
+	 * Same with http proxy port.
+	 *
+	 * If a {@link ClientTransport} instance already has proxy set via {@link ClientTransport#proxy(Consumer)}
+	 * this method will delete the previous proxy settings completely
```

Comment:
should it be clarified that since a new instance is created, the old instance isn't really affected?

Reply:
updated docs to clarify the method creates a copy and the old instance is not affected.